image/oci.go

```diff
@@ -12,12 +12,18 @@ import (
 	"github.com/pkg/errors"
 )
 
+type descriptorOCI1 struct {
```

Comment:
Historically probably because `image/oci.go` was created by copy&pasting `image/docker_schema2.go`, so it ended up using our `descriptor` copy.

I would be just fine with switching to upstream types as well, but it is not a blocker for this PR for me.